### PR TITLE
Jay/oauth token

### PR DESCRIPTION
**Pull request by FWJay**
add authentication/refresh request

### CODE REVIEW

File: lib/harmony/service/oauth_token_request.rb [hunk 1]
```diff
@@ -0,0 +1,3 @@
+class Harmony::Service::OauthTokenRequest < Harmony::Service::Request
+	attr_accessor :auth_code
```
Comment (FWMatt) [inline lib/harmony/service/oauth_token_request.rb]: Why is this not called access_token?

File: lib/harmony/service/oauth_token_response.rb [hunk 2]
```diff
@@ -0,0 +1,3 @@
+class Harmony::Service::OauthTokenResponse < Harmony::Service::Response
+  attr_accessor :oauth_tokens
```
Comment (FWMatt) [inline lib/harmony/service/oauth_token_response.rb]: Why do we not have access_token and refresh_token here?

File: lib/harmony/service/oauth_token_request.rb (hunk 1 above)
Comment (FWJay) [reply lib/harmony/service/oauth_token_request.rb]: this is the authorization code (token from the first call) to invoke access_credentials

File: lib/harmony/service/oauth_token_response.rb (hunk 2 above)
Comment (FWJay) [reply lib/harmony/service/oauth_token_response.rb]: the request a lot of information beyond access_token and refresh_token, like expiry date and auth type so I send it back as a hash for future proofing it. maybe I should call it access_credentials

**Pull request merged**